compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/PEGraphDecoder.java

```diff
@@ -1217,8 +1217,12 @@ protected void finishInlining(MethodScope is) {
 
         assert exceptionValue == null || exceptionValue instanceof FixedAnchorNode && exceptionValue.predecessor() != null;
 
+        PENonAppendGraphBuilderContext notifyGraphBuilderContext = null;
         for (InlineInvokePlugin plugin : inlineInvokePlugins) {
-            plugin.notifyAfterInline(inlineMethod);
+            if (notifyGraphBuilderContext == null) {
+                notifyGraphBuilderContext = new PENonAppendGraphBuilderContext(methodScope, invokeData.invoke);
```

Comment:
`notifyGraphBuilderContext` can be initialized outside the loop. If you want to init it only when there are some inline plugins installed you can check the length of `inlineInvokePlugins`. Seems cleaner that way.

Reply:
Done